### ISSUE TITLE
`nlp.evaluate` with empty `docs_golds` causes a bug

### ISSUE THREAD

**Issue by tamuhey** [explosion/spaCy#4924]
## How to reproduce the behaviour

```python
nlp = spacy.blank("en")
docs_golds = [("", {})]
nlp.evaluate(docs_golds)
```

Causes the error:

```
        gold_deps = set()
        gold_deps_per_dep = {}
        gold_tags = set()
>       gold_ents = set(tags_to_entities([annot[-1] for annot in gold.orig_annot]))
E       TypeError: 'NoneType' object is not iterable

spacy/scorer.py:239: TypeError
```

## Your Environment

* Operating System: osx
* Python Version Used: 3.7.4
* spaCy Version Used: master
* Environment Information:

**Comment by tamuhey** [comment]
created PR in #4925

**Comment by svlandeg** [comment]
Just as a small tip, if you put something like `Fixes #4924` in your PR description, it will automatically close the original issue when merging the PR, and on top will mention something like `Fixed by #4925` ;-)
